src/adr/0040-quality-levels-definitions.md

```diff
@@ -0,0 +1,62 @@
+# 40. Quality levels definition
+
+Date: 2020-06-21
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+We hereby define as an artifact compiled artifact that external people consumes. This includes:
+
+ - libraries
+ - Mail servers
+ - Extensions for James Mail Servers
+ - Command line tools
+
+We designate as a feature an optional, opt-in behaviour of a James server that can be configured by 
+user willing to rely on it.
+
+James as a project delivers several artifacts, and features. In order for project users to better
+understand the underlying quality of the artifact they use, as well as the level of risk associated,
+we need to better define some quality levels.
+
+## Decision
+
+For a given artifact or feature, by **mature** we mean that:
+
+ - *interfaces* in components need a contract test suite
+ - *interfaces* have several implementations
+ - *implementation* of these interfaces need to pass this contract test suite which provides unit tests
+ - Decent integration tests coverage is needed
+ - Performance tests need to be conducted out
+ - Quality Assurance with external clients needs to be conducted out
+ - known existing production deployments/usages
+ - usable documentation
+
+This is the maximum quality level delivered by the James project. Users should feel confident using these
+artifacts or features.
+
+By **experimental** we designate an artifact or feature not matching yet the above requirements. However some
+active contributors are willing to raise the quality level of this component, and eventually make it 
+mature. Or at least are willing to support users.
+
+Users should have low expectations regarding experimental artifacts or features. They are encouraged to contribute to them 
+in order to raise its quality.
+
+By **unsupported** we mean that an artifact or feature do not match most of the *mature* quality conditions. Active 
```

Comment:
In the referenced email, @chibenwa wrote:

> This is a fair point, but 'quality' differs from 'support' concerns.

I agree with this point, which is why I have trouble with the "unsupported" label.

Reply:
Propose an alternative please. 

I want a term that reflexes the "nothing to expect" concepts. "unsupported" convey this idea quite well to be honest.

Reply:
I wrote in a separate comment below how I think the concepts are a bit mixed up. But again, there comes a point at which a boundary has to be drawn, so even though IMO it is confusing, I leave the decision up to you. :)